samples/application_development/out_of_tree_driver/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# For the sake of demonstration, we add the driver directory as a zephyr module
+# by hand. If your driver is a project that's managed by west, you can remove this line.
+list(APPEND ZEPHYR_EXTRA_MODULES
+  ${CMAKE_CURRENT_SOURCE_DIR}/hello_world_module
+  )
+
+cmake_minimum_required(VERSION 3.13.1)
+
+# Add an absolute directory path to the CMake variable
+# SYSCALL_INCLUDE_DIRS. This ensures that the syscall machinery will
+# be able to find the module's syscalls.
+list(APPEND SYSCALL_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/hello_world_module/zephyr)
```

Comment:
if we intend for apps to be adding stuff to SYSCALL_INCLUDE_DIRS maybe we should define a helper macro for doing so.

`zephyr_syscalls_add_include_directory(...path...)` or some such perhaps?

Reply:
Unfortunately, this function wouldn't be available as boilerplate is not sourced yet.

Setting variables before boilerplate.cmake and from CLI is well established so this should be OK.

Reply:
Will update docs

Reply:
docs updated